java/src/main/native/src/PackedColumnMetadataJni.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cudf_jni_apis.hpp"
+
+namespace {
+
+#define PACKED_COLUMN_META_CLASS "ai/rapids/cudf/PackedColumnMetadata"
+#define PACKED_COLUMN_META_FACTORY_SIG(param_sig) "(" param_sig ")L" PACKED_COLUMN_META_CLASS ";"
+
+jclass Packed_columns_meta_jclass;
+jmethodID From_packed_column_meta_method;
+
+} // anonymous namespace
+
+namespace cudf {
+namespace jni {
+
+bool cache_packed_column_meta_jni(JNIEnv *env) {
+  jclass cls = env->FindClass(PACKED_COLUMN_META_CLASS);
+  if (cls == nullptr) {
+    return false;
+  }
+
+  From_packed_column_meta_method =
+      env->GetStaticMethodID(cls, "fromPackedColumnMeta", PACKED_COLUMN_META_FACTORY_SIG("J"));
+  if (From_packed_column_meta_method == nullptr) {
+    return false;
+  }
+
+  // Convert local reference to global so it cannot be garbage collected.
+  Packed_columns_meta_jclass = static_cast<jclass>(env->NewGlobalRef(cls));
+  if (Packed_columns_meta_jclass == nullptr) {
+    return false;
+  }
+  return true;
+}
+
+jobject packed_column_metadata_from(JNIEnv *env, std::unique_ptr<std::vector<uint8_t>> meta) {
```

Comment:
Why do we create a jobect and do all of the reflection/etc if we could just return a jlong to java and let it call the constructor in java?

Reply:
yeap that is much easier, fixed.